language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala

```diff
@@ -338,13 +338,17 @@ class DottyLanguageServer extends LanguageServer
     implicit val ctx = driver.currentCtx
 
     val pos = sourcePosition(driver, uri, params.getPosition)
-    val tp = Interactive.enclosingType(driver.openedTrees(uri), pos)
+    val trees = driver.openedTrees(uri)
+    val tp = Interactive.enclosingType(trees, pos)
     val tpw = tp.widenTermRefExpr
 
     if (tpw == NoType) new Hover
     else {
-      val str = tpw.show.toString
-      new Hover(List(JEither.forLeft(str)).asJava, null)
+      import dotty.tools.dotc.core.Comments._
+      val symbol = Interactive.enclosingSourceSymbol(trees, pos)
+      val docComment = ctx.docCtx.flatMap(_.docstring(symbol))
+      val markedString = docMarkedString(docComment, tpw.show.toString)
+      new Hover(List(JEither.forRight(markedString)).asJava, null)
```

Comment:
What if you return two marked strings? One for the docstring and one for the info. The `textDocument/hover` response can return `MarkedString[]` arrays https://microsoft.github.io/language-server-protocol/specification#textDocument_hover

Reply:
Thanks for the suggestion! I've changed it to return 2 different `MarkedString`s (one with the doc, the other with the type info). Unfortunately, this doesn't improve much the layout on VSCode: it simply adds an horizontal delimiter between the two hovers, but you still need to scroll.

I've inverted the order of the two `MarkedString`.